### PR TITLE
insync 3.8.7.50516 -> 3.9.1.60010 | insync-emblem-icons: init at 3.8.7.50516 | insync-nautilus: fixes

### PR DESCRIPTION
**Pull request by hellwolf**
## Description of changes

- insync: 3.8.7.50516 -> 3.9.0.60000-
- insync-emblem-icons: init at 3.8.7.50516
  - insync-nautilus: use insync-emblem-icons

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/in/insync/package.nix [hunk 1]
```diff
@@ -7,38 +7,26 @@
...
-      "https://cdn.insynchq.com/builds/linux/insync-emblem-icons_${version}_all.deb"
-      "https://web.archive.org/web/20240409081214/https://cdn.insynchq.com/builds/linux/insync-emblem-icons_${version}_all.deb"
-    ];
-    hash = "sha256-uALaIxETEEkjDTx331uIsb4VswWk2K0dGuDMYH8v5U8=";
-  });
+  # Find a binary from https://www.insynchq.com/downloads/linux
+  version = "3.9.0.60000";
+  ubuntu-dist = "trixie_amd64";
   insync-pkg = stdenvNoCC.mkDerivation {
     name = "${pname}-pkg-${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/in/insync/package.nix:21]:
```suggestion
    pname = "${pname}-pkg";
    inherit version;
```
we should properly set pname+version

File: pkgs/by-name/in/insync-emblem-icons/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchurl
+, dpkg
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "insync-emblem-icons";
```
Comment (SuperSandro2000) [inline pkgs/by-name/in/insync-emblem-icons/package.nix:8]:
```suggestion
  pname = "insync-emblem-icons";
```

File: pkgs/by-name/in/insync-emblem-icons/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    urls = [
+      "https://cdn.insynchq.com/builds/linux/insync-emblem-icons_${finalAttrs.version}_all.deb"
+      "https://web.archive.org/web/20240409081214/${builtins.elemAt urls 0}"
+    ];
+    hash = "sha256-uALaIxETEEkjDTx331uIsb4VswWk2K0dGuDMYH8v5U8=";
+  };
+
+  nativeBuildInputs = [ dpkg ];
+
+  unpackPhase = ''
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply pkgs/by-name/in/insync-emblem-icons/package.nix:21]: done

File: pkgs/by-name/in/insync/package.nix [hunk 4]
```diff
@@ -47,24 +35,23 @@ let
...
-      wayland
+      nss
       libthai
-      libsForQt5.qt5.qtvirtualkeyboard
-    ];
+    ] ++ (with libsForQt5; [
+      qt5.qtvirtualkeyboard
+    ]);
 
     unpackPhase = ''
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply pkgs/by-name/in/insync/package.nix:46]: done

File: pkgs/by-name/in/insync-emblem-icons/package.nix (hunk 2 above)
Comment (hellwolf) [reply pkgs/by-name/in/insync-emblem-icons/package.nix]: done

File: pkgs/by-name/in/insync/package.nix (hunk 1 above)
Comment (hellwolf) [reply pkgs/by-name/in/insync/package.nix]: done

**Comment by hellwolf** [comment]
@drupol  could you have a look again?

**Comment by hellwolf** [comment]
I have just upgraded it to 3.9.1, can any of you please review again and maybe merge? @SuperSandro2000 @drupol

**Review by drupol — APPROVED** [review]

**Pull request merged**